sdk/core/core-rest-pipeline/src/util/tokenCycler.ts

```diff
@@ -183,23 +195,31 @@ export function createTokenCycler(
     return refreshWorker as Promise<AccessToken>;
   }
 
-  return async (tokenOptions: GetTokenOptions): Promise<AccessToken> => {
-    //
-    // Simple rules:
-    // - If we MUST refresh, then return the refresh task, blocking
-    //   the pipeline until a token is available.
-    // - If we SHOULD refresh, then run refresh but don't return it
-    //   (we can still use the cached token).
-    // - Return the token, since it's fine if we didn't return in
-    //   step 1.
-    //
-
-    if (cycler.mustRefresh) return refresh(tokenOptions);
-
-    if (cycler.shouldRefresh) {
-      refresh(tokenOptions);
-    }
+  return {
+    get cachedToken(): AccessToken | undefined {
+      return token || undefined;
```

Comment:
a bit amused that we have to do `|| undefined` here, since we are using `null` as one sentinel and `undefined` as another. Can't we just return `null` instead?